File: supervision-client/src/app/shared/technical/reporting/string-search-criteria.ts
import { SearchCriteria } from "./search-criteria";
import { ReportOperator } from "./report-operator";

export class StringSearchCriteria implements SearchCriteria {
    key: string;
    operator: ReportOperator;
    value: string;
    criteriaType: string = 'string';

    public constructor(key: string, operator: ReportOperator, value: string) {
        this.key = key;
        this.operator = operator;
        this.value = value;
    }

    getValue(): string {
        return this.value;
    }
}

File: supervision-server-data/src/main/java/supervision/server/reporting/command/StringSearchCriteria.java
package supervision.server.reporting.command;

import java.time.LocalDate;
import java.time.format.DateTimeParseException;

import lombok.Data;
import lombok.NonNull;

/**
 * {@code @Data} annotation creates getter and setter.
 * A constructor with all fields set as {@code @NonNull} will also be created
 * 
 * @author cyrilg
 *
 */
@Data
public class StringSearchCriteria implements SearchCriteria{
	
	@NonNull
	private String key;
	@NonNull
	private ReportOperator operator;
	@NonNull
	private String value;

	/**
	 * 
	 * 
	 * @date 01/08/2017
	 * @author Cyril Gambis
	 * @return
	 */
	public LocalDate getLocalDateValue() {
		LocalDate result = null;
		if (value != null && value instanceof String) {
			try {
				result = LocalDate.parse(value);
			} catch (DateTimeParseException e) {
				// Do nothing, result is null
			}
		}
		return result;
	}
	
}
